pkg/auth/asynccache.go

```
@@ -27,13 +33,25 @@ func NewAsyncCache[T any](ctx context.Context, reloadPeriod time.Duration, cachi
 		cachingFunc:  cachingFunc,
 	}
 
-	item, err := cachingFunc(ctx)
+	err := wait.PollUntilContextTimeout(
+		ctx,
+		initializationRetryInterval,
+		initializationTimeout,
+		initializeImmediately,
+		func(ctx context.Context) (bool, error) {
+			item, err := cachingFunc(ctx)
+			if err != nil {
+				klog.V(4).Infof("failed to setup an async cache - retrying in %s", initializationRetryInterval)
+				return false, err
```

won't returning an error make the loop fail early, as it would've previously?